docs/design/proposed/interactive-mode.md

```diff
@@ -0,0 +1,64 @@
+# Interactive mode for `dotnet new` instanciation
+
+For .NET 7 `dotnet new` invested heavily in improving the UX of the tool through a comprehensive and data driven tab-completion feature. This feature, when enabled, helps users a) correctly instantiate templates and b) acts as education about what options a template supports.
+
+For .NET 8 we should continue investing in the UX of the `dotnet new` command to make it more interactive and easier to use. This document outlines a proposal for a new interactive mode for `dotnet new` that will guide users in a workflow-style experience, prompting for inputs where required.  This brings the .NET CLI templating experience up to par with user expectations in today's world of modern CLI experiences.
+
+## Goals
+
+* Provide a guided experience for users to provide values for symbols (and other items like the project name)
+* Enforce required symbols, but allow for skipping optional values
+* Support conditional symbols
+* Only provide the interactive experience when a user has entered a command (i.e. isatty is true)
+* Where possible, provide an attractive and informative prompting experience through the use of ANSI control codes and terminal formatting. When the environment doesn't support it, use 'boring' formatting.
+
+## Non-Goals
+
+* Provide interactivity to the entire .NET CLI
+* Extend interactivity to commands like `dotnet new search` or `dotnet new list` (i.e. interactive filtering)
+
+## Workflow
+
+When a user invokes a template, the command should evaluate the user-provided parameters and determine if it should enter interactive mode. This is done by checking if the command is being invoked from a terminal (i.e. isatty is true) and if the command has any symbols that are of type `parameter` and are required _while also having required symbols that are not explicitly provided_. If both of these conditions are true, the command should enter interactive mode.
+
+When interactive mode is entered, the command should prompt the user for each required symbol. If the symbol has a default value, it should be displayed as the default value with some formatting (for example, grayed out text). If the symbol has a description, it should be displayed as part of a prompt. If the symbol has a datatype, the command should validate the input and prompt the user again if the input is invalid. At each stage of the prompt, the user should be able to press a key to skip the prompt and use the default value (if one exists). At each stage of the prompt, the user should be able to press a key to skip all other prompts and use their default values (if they exist). Finally, at each stage of the prompt the user should be able to press a key and quit the entire operation.
+
```

Comment:
Yes - we should use the implicit default as the 'default value' in this case

Reply:
Ok

Should we then possibly reword `If the symbol has a default value` to something like `The symbol default (or implicit default) value should allways be displayed`.

It's a nitpick - but just so that this doesn't slip.

Reply:
done!